app/sprites.py

```diff
@@ -300,28 +344,235 @@ def move(self):
+            Slow()
+            Accelerate()
+
+    def move(self):
+        if self.rect.centerx > 0 and self.rect.centery > 0:
+            self.surf = self.image_right
+            self.rect.move_ip(-self.speed, -self.speed)
+
+    def kill(self) -> None:
+        super().kill()
+        self.engine.score += self.score_points
+
+
+class Rock(EngineMovingSprite):
+    def __init__(self, owner_center):
+        super().__init__()
+        self.engine.add_to_group(self, "rocks")
+        self.engine.add_to_group(self, "flammable")
+        self.surf = pygame.image.load("images/rock.png").convert_alpha()
+        self.rect = self.surf.get_rect(
+            center=self.get_self_center(owner_center)
+        )
+
+    def get_self_center(self, owner_center):
+        lines = self.get_line_rock_placed(owner_center)
+        return (
+            lines[0] * DEFAULT_OBJ_SIZE + self.surf.get_width() // 2,
+            lines[1] * DEFAULT_OBJ_SIZE + self.surf.get_height() // 2,
+        )
+
+    @staticmethod
+    def get_line_rock_placed(owner_center):
+        width = owner_center[0] // DEFAULT_OBJ_SIZE
+        height = owner_center[1] // DEFAULT_OBJ_SIZE
+        return width, height
+
+
+class Effect(EngineMovingSprite, ABC):
+    def __init__(self):
+        super().__init__()
+        self.position = self.generate_position()
+        self.surf = pygame.Surface((20, 10))
+        self.rect = self.surf.get_rect(center=self.position)
+
+    @abstractmethod
+    def update(self):
+        pass
+
+    @abstractmethod
+    def collisions_handling(self):
+        pass
+
     @staticmethod
     def generate_position() -> tuple:
-        delta = 50
-        direction = random.choice(["left", "top", "right", "bottom"])
-        if direction == "left":
-            width = -delta
-            height = random.randint(0, SCREEN_HEIGHT)
+        width = random.randint(0, SCREEN_WIDTH)
+        height = random.randint(0, SCREEN_HEIGHT)
+        return width, height
 
-        if direction == "top":
-            width = random.randint(0, SCREEN_WIDTH)
-            height = -delta
 
-        if direction == "right":
-            width = SCREEN_WIDTH + delta
-            height = random.randint(0, SCREEN_HEIGHT)
+class Healing(Effect):
+    def __init__(self):
+        super().__init__()
+        self.engine.add_to_group(self, "Healing")
+        self.surf = pygame.image.load(
+            "images/healing_effect.png").convert_alpha()
 
-        if direction == "bottom":
-            width = random.randint(0, SCREEN_WIDTH)
-            height = SCREEN_HEIGHT + delta
+    def update(self):
+        self.collisions_handling()
 
-        return width, height, direction
+    def collisions_handling(self):
+        player = self.engine.groups["player"].sprites()[0]
+        if pygame.sprite.spritecollideany(self, self.engine.groups["player"]):
+            player.change_health(200)
+            self.kill()
+
+    def kill(self) -> None:
+        super().kill()
```

Comment:
Why did you override the method `kill` if you only call for `super().kill()`? It's redundant.